lib/Page.js

```diff
@@ -775,20 +775,21 @@ class Page extends EventEmitter {
    * @param {!Array<string>} values
    */
   async select(selector, ...values) {
-    await this.$eval(selector, (element, values) => {
+    return await this.$eval(selector, (element, values) => {
       if (element.nodeName.toLowerCase() !== 'select')
         throw new Error('Element is not a <select> element.');
-
       const options = Array.from(element.options);
-
+      const filteredValues = values.filter(value => options.some(option => option.value === value));
       if (element.multiple) {
         for (const option of options)
           option.selected = values.includes(option.value);
       } else {
-        element.value = values.shift();
+        filteredValues.splice(1);
+        element.value = filteredValues[0];
       }
       element.dispatchEvent(new Event('input', { 'bubbles': true }));
       element.dispatchEvent(new Event('change', { 'bubbles': true }));
+      return Promise.all(filteredValues);
```

Comment:
Promise.all doesn't make sense to me here: there are no promises inside the array

Reply:
I thought I would use Promise.all() so that the function would be thenable. Can you .then() a function when it doesn't return a promise?